src/MicrosoftTeamsMessage.php

```diff
@@ -0,0 +1,378 @@
+     *
+     * @param string $name
+     * @param string $value
+     * @param string|int $sectionId - in which section to put the property, defaults to standard_section
+     *
+     * @return MicrosoftTeamsMessage $this
+     */
+    public function fact(string $name, string $value, $sectionId = 'standard_section'): self
+    {
+        $newFact = compact('name', 'value');
+        $this->payload['sections'][$sectionId]['facts'][] = $newFact;
+
+        return $this;
+    }
+
+    /**
+     * Add an image to a section.
+     *
+     * @param string $imageUri - The URL to the image.
+     * @param string $title - A short description of the image. Typically, title is displayed in a tooltip as the user hovers their mouse over the image
+     * @param string|int $sectionId - in which section to put the property, defaults to standard_section
+     *
+     * @return MicrosoftTeamsMessage $this
+     */
+    public function image(string $imageUri, string $title = '', $sectionId = 'standard_section'): self
+    {
+        $newImage = [
+            'image' => $imageUri,
+            'title' => $title,
+        ];
+        $this->payload['sections'][$sectionId]['images'][] = $newImage;
+
+        return $this;
+    }
+
+    /**
+     * Add a hero image to a section.
+     *
+     * @param string $imageUri - The URL to the image.
+     * @param string $title - A short description of the image. Typically, title is displayed in a tooltip as the user hovers their mouse over the image
+     * @param string|int $sectionId - in which section to put the property, defaults to standard_section
+     *
+     * @return MicrosoftTeamsMessage $this
+     */
+    public function heroImage(string $imageUri, string $title = '', $sectionId = 'standard_section'): self
+    {
+        $newImage = [
+            'image' => $imageUri,
+            'title' => $title,
+        ];
+        $this->payload['sections'][$sectionId]['heroImage'] = $newImage;
+
+        return $this;
+    }
+
+    /**
+     * Additional options to pass to message payload object.
+     *
+     * @param array $options
+     * @param string|int $sectionId - optional in which section to put the property
+     *
+     * @return MicrosoftTeamsMessage $this
+     */
+    public function options(array $options, $sectionId = null): self
+    {
+        if ($sectionId) {
+            $this->payload['sections'][$sectionId] = array_merge($this->payload['sections'][$sectionId], $options);
+        }
+        $this->payload = array_merge($this->payload, $options);
+
+        return $this;
+    }
+
+    /**
+     * Recipient's webhook url.
+     *
+     * @param $webhookUrl - url of webhook
+     *
+     * @throws CouldNotSendNotification
+     *
+     * @return MicrosoftTeamsMessage $this
+     */
+    public function to(?string $webhookUrl): self
+    {
+        if (! $webhookUrl) {
+            throw CouldNotSendNotification::microsoftTeamsWebhookUrlMissing();
```

Comment:
Not sure why this takes a nullable string, then throws an exception?

Reply:
Yeah I already faced that problem and changed it to a nullable string due to the following case:

I you forget to define the `to()` method or in my case the config or env setting in the to method (e.g. `->to(config('services.teams.subscriptions_url'))` than you get the following error:
```
Argument 1 passed to NotificationChannels\MicrosoftTeams\MicrosoftTeamsMessage::to() must be of the type string, null given
```
Which is not a very meaningful error imho if you don't know the architecture of the package. So I made it nullable and throw a more meaningful Exception if the param in to() is empty:
```
(NotificationChannels\\MicrosoftTeams\\Exceptions\\CouldNotSendNotification(code: 0):
Microsoft Teams webhook url is missing. Please add it as param over the MicrosoftTeamsMessage::to($url) method or return it in the notifiable model by providing the method Model::routeNotificationForMicrosoftTeams().
```

I can change it back, but this was just my thinking. What do you think, maybe there is a better approach?

Reply:
Ahh, I understand (I didn't see that you are calling this method with the result of the routing on the notifiable model)
All good, thanks for the explaination!